src/app.js

```diff
@@ -0,0 +1,195 @@
+/*global $:false, Game, Frame */
+
+$(document).ready(function() {
+  var game,
+      gameNumber = 0,
+      rollValue;
+
+  startNewGame();
+
+  $( 'body' ).on( 'click', '#startNewGame', function(event) {
+   $( '#userInterface > p' ).html( 'Click the number of pins to knock down' ); 
+   game = new Game();
+   gameNumber++;
+   drawButtons();
+   drawScoreCard();
+  });
+
+  $( 'body' ).on('click', '.btn', function(event) {
+    rollValue = $(this).attr('data-value');
+
+    game.roll( parseInt(rollValue) );
+    updateScore();
+    game.isGameOver() ? startNewGame() : drawButtons();
+  })
+
+
+  function startNewGame() {
+    var newGame = '<button id="startNewGame" type="button" ' +
+                  'data-value="Play a new Game">Play a new Game</button>';
+   $( '#buttonList' ).empty();
+   $( '#userInterface > p' ).html(newGame); 
+  }
+
+  function drawButtons() {
+    var i, btn, numButtons,
+        currentFrame = game.currentFrame();
+    
+    if( currentFrame.isLastFrame() && 
+       ( currentFrame.isStrike() || 
+         currentFrame.pins === 20 ||
+         currentFrame.isSpare() ) ) {
+      numButtons = 10; 
+    } else {
+      numButtons = 10 - currentFrame.pins;
+    }
+
+    $( '#buttonList' ).empty();
+
+    for( i = 0; i <= numButtons; i++ ) {
+      btn  = "<button id='btn" + i + "' type='button' class='btn' " + 
+        "data-value='" + i + "'>" + i + "</button>";
+      $( '#buttonList' ).append(btn);
+    }
+  }
+
+  function updateScore() {
+    game.frames.forEach( function( frame, index ) {   
```

Comment:
This function has too many statements. (12)<br>This function's cyclomatic complexity is too high. (13)